backend/env/env.go

```diff
@@ -11,20 +11,43 @@ import (
 // NewBackend creates a configuration loader that loads from the environment.
 // If the key is not found, this backend tries again by turning any kebabcase key to snakecase and
 // lowercase letters to uppercase.
-func NewBackend() backend.Backend {
+func NewBackend(fns ...opt) backend.Backend {
```

Comment:
Technically I'm afraid this is a backwardly incompatible change, as someone might have assigned the function to a variable of type `func() backend.Backend`.

Also, I don't think it's a good idea in general to have unexported functions that use unexported types (it makes the godoc hard to understand and means you can't declare variables of those types).